pkg/prowloader/bigqueryjobs.go

```diff
@@ -0,0 +1,111 @@
+		// job START time, but we'll subtract another 10 minutes just in case there were upload delays to gcs/bigquery:
+		lastProwJobRun = lastProwJobRun.Add(-10 * time.Minute)
+	}
+	log.Infof("Loading prow jobs from bigquery completed since: %s", lastProwJobRun.UTC().Format(time.RFC3339))
+
+	// NOTE: casting a couple datetime columns to timestamps, it does appear they go in as UTC, and thus come out
+	// as the default UTC correctly.
+	// Annotations and labels can be queried here if we need them.
+	query := pl.bigQueryClient.Query(`SELECT 
+			prowjob_job_name, 
+			prowjob_state, 
+			prowjob_build_id, 
+			prowjob_type, 
+			prowjob_cluster, 
+			TIMESTAMP(prowjob_start) AS prowjob_start_ts, 
+			TIMESTAMP(prowjob_completion) AS prowjob_completion_ts ` +
+		"FROM `ci_analysis_us.jobs` " +
+		`WHERE TIMESTAMP(prowjob_completion) > @queryFrom ORDER BY prowjob_start_ts`)
+	query.Parameters = []bigquery.QueryParameter{
+		{
+			Name:  "queryFrom",
+			Value: lastProwJobRun,
+		},
+	}
+	it, err := query.Read(context.TODO())
+	if err != nil {
+		errs = append(errs, err)
+		log.WithError(err).Error("error querying jobs from bigquery")
+		return []prow.ProwJob{}, errs
+	}
+
+	prowJobs := []prow.ProwJob{}
+	for {
+		bqjr := bigqueryProwJobRun{}
+		err := it.Next(&bqjr)
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.WithError(err).Error("error parsing prowjob from bigquery")
+			errs = append(errs, errors.Wrap(err, "error parsing prowjob from bigquery"))
+			continue
+		}
+
+		// Convert to a prow.ProwJob:
+		pj := prow.ProwJob{
+			Spec: prow.ProwJobSpec{
+				Type:    bqjr.Type,
+				Cluster: bqjr.Cluster,
+				Job:     bqjr.JobName,
+			},
+			Status: prow.ProwJobStatus{
+				StartTime:      bqjr.StartTime,
+				CompletionTime: &bqjr.CompletionTime,
+				State:          prow.ProwJobState(bqjr.State),
+				URL:            bqjr.GetURL(),
+				BuildID:        bqjr.BuildID,
+			},
+		}
+		prowJobs = append(prowJobs, pj)
+	}
+
+	log.Infof("found %d jobs in bigquery since last import (roughly)", len(prowJobs))
+
+	return prowJobs, errs
+}
+
+// bigqueryProwJobRun is a transient struct for processing results from the bigquery jobs table.
+// Ultimately just used to convert to a prow.ProwJob.
+type bigqueryProwJobRun struct {
+	JobName        string    `bigquery:"prowjob_job_name"`
+	State          string    `bigquery:"prowjob_state"`
+	BuildID        string    `bigquery:"prowjob_build_id"`
+	Type           string    `bigquery:"prowjob_type"`
+	Cluster        string    `bigquery:"prowjob_cluster"`
+	StartTime      time.Time `bigquery:"prowjob_start_ts"`
+	CompletionTime time.Time `bigquery:"prowjob_completion_ts"`
+}
+
+// GetURL returns the guessed URL for the prow job using it's name and build ID.
+func (jr bigqueryProwJobRun) GetURL() string {
+	return fmt.Sprintf("https://prow.ci.openshift.org/view/gcs/origin-ci-test/logs/%s/%s", jr.JobName, jr.BuildID)
```

Comment:
This won't work for presubmits, the URL is different.

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/1403/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws-ovn/1598319526313725952/

Could we just get testplatform to store the URL in their table?

Reply:
Looked like it was working for some but I see now it's not consistent at the very least. Working on getting it into the bigquery db.

/hold

Reply:
I got the importer now setting prowjob_url, and updated this code to use it. Includes a filter for not null as that breaks serialization if we pickup any with out it set. I'll keep an eye on the db to make sure none are slipping in still, but by this time this rolls out, everything should have a URL set.